Kernel/Graphics/Bochs/GraphicsAdapter.cpp

```diff
@@ -109,8 +109,9 @@ UNMAP_AFTER_INIT BochsGraphicsAdapter::BochsGraphicsAdapter(PCI::DeviceIdentifie
     m_framebuffer_console = Graphics::ContiguousFramebufferConsole::initialize(PhysicalAddress(PCI::get_BAR0(pci_device_identifier.address()) & 0xfffffff0), 1024, 768, 1024 * sizeof(u32));
     GraphicsManagement::the().set_console(*m_framebuffer_console);
 
-    // Note: If we use VirtualBox graphics adapter (which is based on Bochs one), we need to use IO ports
-    if (pci_device_identifier.hardware_id().vendor_id == 0x80ee && pci_device_identifier.hardware_id().device_id == 0xbeef)
+    // Note: If we use Bochs or VirtualBox graphics adapter (which is based on Bochs one), we need to use IO ports
```

Comment:
This does not look life. It's almost unreadable - I assume the list will grow in time... so better be prepared (the compiler will optimize it the same way).

How about:
``` C++
auto vendor_id = pci_device_identifier.hardware_id().vendor_id;
auto device_id = pci_device_identifier.hardware_id().device_id;
auto revision_id = pci_device_identifier.revision_id();

if ((vendor_id == PCI::VendorID::QEMUOld && device_id==0x1111 && revision_id = 0) ||
   (vendor_id == 0x80ee && device_id == 0xbeef) {
...
}
```

Or another alternative:
``` C++

auto vendor_id = pci_device_identifier.hardware_id().vendor_id;
auto device_id = pci_device_identifier.hardware_id().device_id;
auto revision_id = pci_device_identifier.revision_id();

auto is_qemu = vendor_id == PCI::VendorID::QEMUOld && device_id == 0x1111 && revision_id() == 0;
auto is_bochs  = vendor_id == 0x80ee && device_id == 0xbeef;

if (is_qemu || is_bochs) {
....
}
```

(not to mention that vendor `0x80ee` can be saved as a constant...).  Maybe you can inline this inside a function, to clean up code?

Reply:
Thanks for the review. I have adopted this convention and replaced magic numbers with constants.